### ISSUE TITLE
Allow opening the addon option page from the website and scroll to a study card

### ISSUE THREAD

**Issue by Dexterp37** [mozilla-ion/ion-core-addon#138]
@knowtheory mentioned that we might want to allow the website to perform a new kind of interaction: if the user already has the addon installed, instead of showing the "Install Addon" button, we could let user browse the list of studies. They would then, from the website, click on "Join Study" (or something like that). The button would send a message to the addon, which would open the options page and scroll to the study card.

**Comment by rhelmer** [comment]
This is really cool! Note that for Firefox we could register a protocol handler to make this easy:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/protocol_handlers

Chrome does not support this though but we could simulate it using https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/webRequest 

Note that I don't think there is a great way to prevent any random website from causing this action, that doesn't seem harmful to me offhand but something to keep in mind.

**Comment by knowtheory** [comment]
oops, i just filed #140 about this.

**Comment by knowtheory** [comment]
The idea I have wouldn't even require protocol handlers, although that's a super interesting consideration for more general way to provide URIs for studies!

**Comment by Dexterp37** [comment]
> 
> 
> oops, i just filed #140 about this.

No worries, let's close this in favour of that!

**Comment by Dexterp37** [comment]
> 
> 
> This is really cool! Note that for Firefox we could register a protocol handler to make this easy:
> https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/protocol_handlers

I noticed I forgot to respond to this, sorry! During our initial explorations, we tested protocol handlers as well. Unfortunately, this has a few limitations (especially the origin limitation described in the [Exceptions](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler) section) that make this not very suitable for us.

A simpler, cross-browser approach could be to still use the content-script, and provide a specific message.
